nixos/modules/services/matrix/draupnir.nix

```
@@ -0,0 +1,247 @@
+          };
+
+          automaticallyRedactForReasons = mkOption {
+            type = types.listOf types.str;
+            default = [ ];
+            description = lib.mdDoc ''
+              A list of reasons for which the bot will automatically redact messages.
+            '';
+          };
+          managementRoom = mkOption {
+            type = types.str;
+            default = "#moderators:example.org";
+            description = lib.mdDoc ''
+              The room ID or alias where moderators can use the bot's functionality.
+
+              The bot has no access controls, so anyone in this room can use the bot - secure this room!
+
+              Warning: When using a room alias, make sure the alias used is on the local homeserver!
+              This prevents an issue where the control room becomes undefined when the alias can't be resolved.
+            '';
+          };
+
+          protectedRooms = mkOption {
+            type = types.listOf types.str;
+            default = [ ];
+            example = literalExpression ''
+              [
+                "https://matrix.to/#/#yourroom:example.org"
+                "https://matrix.to/#/#anotherroom:example.org"
+              ]
+            '';
+            description = lib.mdDoc ''
+              A list of rooms to protect (matrix.to URLs).
+              These can also be configured interactively.
+            '';
+          };
+          #endregion
+        };
+      };
+    };
+    #endregion
+  };
+  #endregion
+
+  #region Service configuration
+  config = mkIf cfg.enable {
+    assertions = [
+      {
+        assertion = !(cfg.pantalaimon.enable && cfg.pantalaimon.passwordFile == null);
+        message = "Specify pantalaimon.passwordFile";
+      }
+      {
+        assertion = !(cfg.pantalaimon.enable && cfg.accessTokenFile != null);
+        message = "Do not specify accessTokenFile when using pantalaimon";
+      }
+      {
+        assertion = !(!cfg.pantalaimon.enable && cfg.accessTokenFile == null);
+        message = "Specify accessTokenFile when not using pantalaimon";
+      }
+    ];
+
+    services.pantalaimon-headless.instances."draupnir" = mkIf cfg.pantalaimon.enable
+      {
+        homeserver = cfg.homeserverUrl;
+      } // cfg.pantalaimon.options;
+
+    systemd.services.draupnir = {
+      description = "Draupnir - a moderation tool for Matrix";
+      requires = optionals (cfg.pantalaimon.enable) [
+        "pantalaimon-draupnir.service"
+      ];
+      wants = [
+        "network-online.target"
+        "matrix-synapse.service"
+        "conduit.service"
+        "dendrite.service"
+      ];
+      after = [
+        "network-online.target"
+        "matrix-synapse.service"
+        "conduit.service"
+        "dendrite.service"
+      ];
+      wantedBy = [ "multi-user.target" ];
+
+      serviceConfig = {
+        ExecStart = utils.escapeSystemdExecArgs ([
+          (lib.getExe pkgs.draupnir)
+          "--draupnir-config" "${configFile}"
```

```suggestion
          "--draupnir-config" configFile
```

I think we don't need the interpolation here.